api/model/portal_catalogue.go

```diff
@@ -7,53 +7,116 @@ type APICatalogue struct {
 	Email string           `json:"email"`
 }
 
-type CatalogueVersion string
+// All auth_type from dashboard validation:Enum=multiAuth;keyless;basic;hmac;jwt;oauth;openid;mutualTLS;authToken;custom;other
+// +kubebuilder:validation:Enum=keyless;jwt;oauth;authToken
+type AuthType string
 
 type APIDescription struct {
-	Name             string                   `json:"name"`
-	ShortDescription string                   `json:"short_description"`
-	LongDescription  string                   `json:"long_description"`
-	Show             bool                     `json:"show"`
-	APIID            string                   `json:"api_id"`
-	PolicyID         string                   `json:"policy_id"`
-	Documentation    string                   `json:"documentation"`
-	Version          *CatalogueVersion        `json:"version,omitempty"`
-	IsKeyless        bool                     `json:"is_keyless,omitempty"`
-	Config           *PortalModelPortalConfig `son:"config,omitempty"`
-	Fields           map[string]string        `json:"fields,omitempty"`
-	AuthType         string                   `json:"auth_type,omitempty"`
+
+	// Name is the title of the API that you wish to be published to the catalogue
+	Name string `json:"name,omitempty"`
+
+	// AuthType displays as a badge next to the name of the API
+	AuthType AuthType `json:"auth_type,omitempty"`
+
+	// Show toggles visibility of the API in the portal catalogue
+	Show bool `json:"show,omitempty"`
+
+	// TODO: I don't think this is exposed to the default portal templates.
+	ShortDescription string `json:"short_description,omitempty"`
```

Comment:
@buger do you know anything about this field? It seems that the short_description has no effect on visibility in the templates. It seems to have no effect even when configured in the GUI.

Reply:
Most likely not. Try creating API catalog via UI, filling all fields - if this field not used, ignore it.